food-truck-api/src/main/java/food/truck/api/endpoint/ReviewEndpoint.java

```diff
@@ -73,22 +76,26 @@ public Review getTruckReviews(@PathVariable long truckId, @RequestParam Long use
     @Secured({"ROLE_USER"})
     @PostMapping("/reviews/truck/{truckId}")
     public Review postTruckReview(@AuthenticationPrincipal User user, @PathVariable long truckId, @RequestBody PostReviewParams data) {
-        // TODO decide if we want to just silently change the review to match
-        if (!user.getId().equals(data.userId))
-            throw new ResponseStatusException(HttpStatus.FORBIDDEN, "UserID does not match userID of the provided review");
-
         // Ensure truck exists
         Optional<Truck> truck = truckService.findTruckById(truckId);
         if (truck.isEmpty())
             throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Provided truck does not exist");
 
+        // Ensure not rating my own truck
+        Truck t = truck.get();
+        if (user.getId().equals(t.getUserId()))
+            throw new ResponseStatusException(HttpStatus.FORBIDDEN, "Cannot rate your own truck");
+
         // Save review
-        return reviewService.saveReview(data.userId, truck.get(), data.score, data.costRating, data.reviewText);
+        return reviewService.saveReview(user.getId(), truck.get(), data.score, data.costRating, data.reviewText);
     }
 
     @Secured({"ROLE_USER"})
     @DeleteMapping("/reviews/truck/{truckId}")
     public boolean deleteReview(@AuthenticationPrincipal User u, @PathVariable long truckId) {
-        return reviewService.deleteReviewByUser(u, truckId);
+        boolean result = reviewService.deleteReviewByUser(u, truckId);
```

Comment:
This should also be checked if the user owns the review

Reply:
Accidentally posted this on the wrong iteration of this comment:
"Forgot to follow up on this: so the way it fetches the review in the service, it actually doesn't need/can't be authenticated. It fetches the review matching the userId from AuthenticationPrinciple and the provided truckId, so by definition the user can't delete someone else's review unless they're signed in as someone else or defeat our authentication system, neither of which we can't check for."